src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -4219,7 +4219,7 @@ trait Types
         tp2 match {
           case mt2 @ MethodType(params2, res2) =>
             // sameLength(params1, params2) was used directly as pre-screening optimization (now done by matchesQuantified -- is that ok, performancewise?)
-            mt1.isImplicit == mt2.isImplicit &&
+            mt1.isImplicit == mt2.isImplicit && //
```

Comment:
The existing implementation looks suspicious to me, it allows one to override a method without the `override` modifier:

```
scala> class C[T >: Null] { def a(implicit i: T = null): Any = "C"}; class S extends C[String] { def a(i: String) = "S" }
defined class C
defined class S

scala> (new S() : C[String]).a()
res8: Any = S
```

My intuition is that `matchesType` should ignore implicit modifers when determining overriding relationships, and a separate check in refchecks should enforce that the overriding method has the same implicit modiifier (or modifiers!) as the overriden one.